test_scripts/Defects/8_2/3888/common.lua

```diff
@@ -0,0 +1,314 @@
+      local corId = mobSession:SendRPC("RegisterAppInterface", params)
+      actions.getHMIConnection():ExpectNotification("BasicCommunication.OnAppRegistered", {
+        application = { appName = params.appName } })
+      mobSession:ExpectResponse(corId, { success = true, resultCode = raiResult or "SUCCESS" })
+    end)
+
+  actions.getHMIConnection():ExpectRequest("BasicCommunication.ActivateApp", { appID = actions.getHMIAppId() })
+  :Do(function(_, data)
+      actions.getHMIConnection():SendResponse(data.id, "BasicCommunication.ActivateApp", "SUCCESS", {})
+    end)
+
+  actions.getMobileSession():ExpectNotification("OnHMIStatus",{ hmiLevel = "NONE" }, { hmiLevel = "FULL" })
+  :Times(2)
+
+  actions.getHMIConnection():ExpectRequest("VR.AddCommand", pParams.vrAddCommandRequest or {})
+  :Do(function(_, data)
+      actions.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+  :ValidIf(function(_, data)
+      if data.params.grammarID ~= nil then
+        return true
+      else
+        return false, "grammarID should not be empty"
+      end
+    end)
+  :Times(pParams.vrCommandTimes)
+
+  if additionalExpectations then
+    local cmdId
+    if pParams.vrAddCommandRequest then cmdId = pParams.vrAddCommandRequest.cmdID end
+    additionalExpectations(pParams.vrCommandTimes, cmdId)
+  end
+
+  utils.wait(1000)
+end
+
+function m.deleteInteractionChoiceSet(pChoiceId, pTimes)
+  local cid = actions.getMobileSession():SendRPC("DeleteInteractionChoiceSet", { interactionChoiceSetID = pChoiceId })
+  actions.getHMIConnection():ExpectRequest("VR.DeleteCommand", { cmdID = pChoiceId })
+  :Do(function(_, data)
+      actions.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+  :Times(pTimes)
+  actions.getMobileSession():ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+  actions.getMobileSession():ExpectNotification("OnHashChange")
+end
+
+function m.addSubMenu()
+  local cid = actions.getMobileSession():SendRPC("AddSubMenu", addSubMenuParams)
+  actions.getHMIConnection():ExpectRequest("UI.AddSubMenu")
+  :Do(function(_, data)
+      actions.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+  actions.getMobileSession():ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+  actions.getMobileSession():ExpectNotification("OnHashChange")
+  :Do(function(_, data)
+      m.hashID = data.payload.hashID
+    end)
+end
+
+function m.additionalResumptionContions(pTimes, pCmdId)
+  actions.getHMIConnection():ExpectRequest("UI.AddSubMenu")
+  :Do(function(_, data)
+      actions.getHMIConnection():SendError(data.id, data.method, "REJECTED", "Error response")
+    end)
+  actions.getHMIConnection():ExpectRequest("VR.DeleteCommand", { cmdID = pCmdId })
+  :Do(function(_, data)
+      actions.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+  :Times(pTimes)
+end
+
+local function getPIreqParams()
```

Comment:
```suggestion
local function getPIReqParams()
```
Since all the other functions use camelCase might be good to keep it consistent